firstvoices/backend/tests/test_apis/test_search_api.py

```diff
@@ -181,3 +183,132 @@ def test_invalid_page_sizes(self, page_size, db, mock_search_query_execute):
+                            "site_id": site.id,
+                        },
+                    }
+                ],
+                "total": {"value": 1, "relation": "eq"},
+            }
+        }
+        mock_search_query_execute.return_value = mock_es_results
+
+        with patch(
+            "backend.pagination.SearchPageNumberPagination.apply_search_pagination",
+            return_value=None,
+        ):
+            response = self.client.get(self.get_list_endpoint())
+            data = response.data
+
+            assert response.status_code == 200
+            assert type(data) is list
+            assert (
+                data[0]["searchResultId"] == mock_es_results["hits"]["hits"][0]["_id"]
+            )
+            assert data[0]["entry"]["id"] == str(entry.id)
+
+
+@pytest.mark.django_db
+class TestSiteSearchAPI(BaseSiteContentApiTest):
+    """Remaining tests that cover the site search."""
+
+    API_LIST_VIEW = "api:site-search-list"
+    API_DETAIL_VIEW = "api:site-search-detail"
+
+    @pytest.fixture
+    def mock_search_query_execute(self, mocker):
+        mock_execute = mocker.patch.object(Search, "execute", new_callable=MagicMock)
+
+        return mock_execute
+
+    def test_invalid_category_id(self):
+        site, user = factories.get_site_with_member(
+            site_visibility=Visibility.PUBLIC, user_role=Role.LANGUAGE_ADMIN
+        )
+        self.client.force_authenticate(user=user)
+
+        response = self.client.get(
+            self.get_list_endpoint(site_slug=site.slug) + "?category=xyzCategory"
+        )
+        response_data = json.loads(response.content)
+
+        assert response_data == []
+
+    def test_starts_with_char_param(self, mock_search_query_execute):
+        site, user = factories.get_site_with_member(
+            site_visibility=Visibility.PUBLIC, user_role=Role.LANGUAGE_ADMIN
+        )
+        entry = factories.DictionaryEntryFactory.create(
+            site=site, visibility=Visibility.PUBLIC, type=TypeOfDictionaryEntry.WORD
+        )
+        self.client.force_authenticate(user=user)
+
+        mock_es_results = {
+            "hits": {
+                "hits": [
+                    {
+                        "_index": "dictionary_entries_2023_06_23_06_11_22",
+                        "_id": "QcHg5ogB3WiEloeO9rdy",
+                        "_score": 1.0,
+                        "_source": {
+                            "document_id": entry.id,
+                            "site_id": site.id,
+                            "title": "xyz",
+                        },
+                    }
+                ],
+                "total": {"value": 1, "relation": "eq"},
+            }
+        }
+
+        mock_search_query_execute.return_value = mock_es_results
+
+        response = self.client.get(
+            self.get_list_endpoint(site_slug=site.slug) + "?startsWithChar=x"
+        )
+
+        assert "startsWithChar=x" in response.request["QUERY_STRING"]
```

Comment:
I don't think this is testing anything useful for us (just that you used the correct url on line 311). A better assertion would be something like 

`mock_search_query_execute.assert_called_with(my_expected_value)`